components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -816,6 +821,9 @@ private void prepareAddedRemovedUserLists(Set<String> addedMembers, Set<String>
                 removedMembers.add(memberObject.get(SCIMConstants.RoleSchemaConstants.DISPLAY));
             }
         } catch (UserStoreException e) {
+            if(e.getMessage().equals("Invalid Domain Name")) {
```

Comment:
Doesn't this exception supports error codes? If supported use error code match.

If the error message has a constant defined in userstore layer, use that constant 

Otherwise update as
`"Invalid Domain Name".equals(e.getMessage())`

Reply:
It does not have an error code.